### ISSUE TITLE
New class for Centreon API V2

### ISSUE THREAD

**Issue by BenoitPoulet** [HenriWahl/Nagstamon#812]
Hello,

I have written a new class for Centreon, who is using the new RestAPI called v2.

At this time the class has the same name of the old Centreon class.

I would like to add this new class into Nagstamon, perhaps with another name.
The old class support Centreon until 21.10, but no more.

This class need some change in the Server config page, as the «auto login» seems deprecated and their is no more a way to check the Centreon's version, so user must provide it. (Centreon can change things between versions so we must have a way to know what to do)

![nagstamon_server_conf](https://user-images.githubusercontent.com/2464712/171158051-8326c248-54f0-41a4-b309-39ded69f532a.PNG)

You can find the class here :
https://github.com/BenoitPoulet/Nagstamon/blob/master/Nagstamon/Servers/Centreon.py

No PEP 8 has been applied to this class, i will try to enhanced it later.

Tell me what will be the next step.

Regards.

**Comment by HenriWahl** [comment]
If I understand it right your new version won't work with older Centreons? So maybe it should just be renamed, similar as you did already in the dropbox, something like Centreon API in a CentreonAPI.py file?

**Comment by BenoitPoulet** [comment]
Yes, this new class in not compatible with older Centreon
I renamed the file into : CentreonAPI.py , and set the variable TYPE to "Centreon API"

I assumed this fact : 
Centreon < 22.xx => Centreon.py (old behavior by parsing XML from the UI)
Centreon >= 22.xx => CentreonAPI.py (Full RestAPI implementation)

I asked the Centreon's team about the API stability, they told me that the API will be stable in the same major version, like 22.xx
So we only need to ask the major version to the user.

New file : https://github.com/BenoitPoulet/Nagstamon/blob/master/Nagstamon/Servers/CentreonAPI.py

**Comment by BenoitPoulet** [comment]
Hello, how to move forward ? I will love to see this support implemented soon but writing QT interface code is out of reach for me.

**Comment by HenriWahl** [comment]
@BenoitPoulet just do a pull request and we will see how it will be merged.

**Comment by HenriWahl** [comment]
I added a new branch to integrate the new Centreon API. Addititonally I created a switch at https://github.com/HenriWahl/Nagstamon/blob/integrate_centreon_api/Nagstamon/Servers/Centreon/__init__.py.
@BenoitPoulet this does not work yet for me but maybe this is a prototype how it could work? Can you please have a more Centreonic look at it?

**Comment by BenoitPoulet** [comment]
My thumb means : I will start to work on this this morning.

**Comment by BenoitPoulet** [comment]
I added some code to switch to the API from the 1rst version really usable.

But i can get the class loading properly.

this line do not seems to work :

DEBUG: 2022-06-27 14:31:38.781399 Test Centreon 20.10 FetchURL: https://centreon.eolas.fr/centreon/status.cgi?hostgroup=all&style=hostdetail&hoststatustypes=12&hostprops=262144&limit=0 CGI Data: None

https://github.com/BenoitPoulet/Nagstamon/blob/integrate_centreon_api/Nagstamon/Servers/Centreon/__init__.py

**Comment by HenriWahl** [comment]
At least I can confirm your observation.
Looks like the attempt to re-initialize the module simply fails - there must be another way.

**Comment by HenriWahl** [comment]
It took a while but now I think I have a simple solution - see https://github.com/HenriWahl/Nagstamon/commit/25d350fcae372574dab81df39ae7c39bff4b273f for details.

**Comment by BenoitPoulet** [comment]
Hello Henri,

Thanks ! I tried to find a fix yesterday without finding it.

I will do some checks on my Centreon's platform.
I will come back soon.

**Comment by HenriWahl** [comment]
Yes - I just think we should examine the autologin respectively disable it in GUI if it is not available.
